skopt/learning/gp_kernels.py

```diff
@@ -0,0 +1,186 @@
+from math import sqrt
+
+import numpy as np
+from sklearn.gaussian_process.kernels import Kernel as sk_Kernel
+from sklearn.gaussian_process.kernels import ConstantKernel as sk_ConstantKernel
+from sklearn.gaussian_process.kernels import DotProduct as sk_DotProduct
+from sklearn.gaussian_process.kernels import Exponentiation as sk_Exponentiation
+from sklearn.gaussian_process.kernels import ExpSineSquared as sk_ExpSineSquared
+from sklearn.gaussian_process.kernels import Matern as sk_Matern
+from sklearn.gaussian_process.kernels import Product as sk_Product
+from sklearn.gaussian_process.kernels import RationalQuadratic as sk_RationalQuadratic
+from sklearn.gaussian_process.kernels import RBF as sk_RBF
+from sklearn.gaussian_process.kernels import Sum as sk_Sum
+from sklearn.gaussian_process.kernels import WhiteKernel as sk_WhiteKernel
+
+
+class Kernel(sk_Kernel):
+    """
+    Base class for skopt.gaussian_process kernels.
+    Supports computation of the gradient of the kernel with respect to X
+    """
+    def gradient_X(self, x, Y):
```

Comment:
I find the use of `x`, `X` and `Y` a bit confusing. Coming from `sklearn` land where there is a clear meaning for `X` and `y` already you have to think a few times before adopting to this convention. I'll make a standalone comment for this.